src/Lang/Core.mli

```diff
@@ -53,6 +53,9 @@ type _ typ =
   | TUnit : ktype typ
     (** Unit type *)
 
+  | TUVar : UID.t * 'k kind -> 'k typ
+    (** UVar type kept for pretty printing *)
```

Comment:
Are they really kept for pretty-printing? Or they are just placeholders for unification variables left by type-inference, and makes sense only in REPL. The invariant about this construct should be said explicitly: they might appear only in REPL mode, and once they appear in the program, information about types in Core becomes inaccurate, and all the type-based transformations (including `WellTypedInvariant`) become unavailable.